src/components/PeopleFilters.tsx

```diff
@@ -27,66 +81,56 @@ export const PeopleFilters = () => {
       <div className="panel-block">
         <div className="level is-flex-grow-1 is-mobile" data-cy="CenturyFilter">
           <div className="level-left">
-            <a
-              data-cy="century"
-              className="button mr-1"
-              href="#/people?centuries=16"
-            >
-              16
-            </a>
-
-            <a
-              data-cy="century"
-              className="button mr-1 is-info"
-              href="#/people?centuries=17"
-            >
-              17
-            </a>
-
-            <a
-              data-cy="century"
-              className="button mr-1 is-info"
-              href="#/people?centuries=18"
-            >
-              18
-            </a>
-
-            <a
-              data-cy="century"
-              className="button mr-1 is-info"
-              href="#/people?centuries=19"
-            >
-              19
-            </a>
-
-            <a
-              data-cy="century"
-              className="button mr-1"
-              href="#/people?centuries=20"
-            >
-              20
-            </a>
+            {availableCenturies.map((century) => (
+              <Link
+                data-cy="century"
+                className={classNames('button mr-1', {
+                  'is-info': centuries.includes(century.toString()),
+                })}
+                to={{
+                  search: getSearchWith(searchParams, {
+                    centuries: centuries.includes(century.toString())
+                      ? centuries.filter(ch => century.toString() !== ch)
+                      : [...centuries, century.toString()],
+                  }),
+                }}
+                key={century}
+              >
+                {century}
+              </Link>
+            ))}
```

Comment:
Try to break the code into simpler parts that are easier to read and maintain.
```suggestion
            {availableCenturies.map((century) => {
              const preparedCentury = century.toString();
              const searchQuery = getSearchWith(searchParams, {
                    centuries: centuries.includes(preparedCentury)
                      ? centuries.filter(ch => preparedCentury !== ch)
                      : [...centuries, preparedCentury] });
                      
              return <Link
                        data-cy="century"
                        className={classNames('button', 'mr-1', {
                          'is-info': centuries.includes(century.toString()),
                        })}
                        to={{ search: searchQuery }}
                        key={century}
                      >
                        {century}
                     </Link>
            })}
```